src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/Artifactory/ServerlessArtifactoryRepository.tsx

```diff
@@ -0,0 +1,157 @@
+import { Failure, useGetRepositoriesDetailsForArtifactory } from 'services/cd-ng'
+import type { ProjectPathProps } from '@common/interfaces/RouteInterfaces'
+import { useStrings } from 'framework/strings'
+import { ConfigureOptions } from '@common/components/ConfigureOptions/ConfigureOptions'
+import { EXPRESSION_STRING } from '@pipeline/utils/constants'
+import css from '../../ArtifactConnector.module.scss'
+
+function NoRepositoryResults({ error }: { error: GetDataError<Failure | Error> | null }): JSX.Element {
+  const { getString } = useStrings()
+
+  return (
+    <span className={css.padSmall}>
+      <Text lineClamp={1}>
+        {get(error, 'data.message', null) || getString('pipeline.artifactsSelection.errors.noRepositories')}
+      </Text>
+    </span>
+  )
+}
+
+interface ServerlessArtifactoryRepositoryProps {
+  expressions: string[]
+  isReadonly?: boolean
+  allowableTypes: MultiTypeInputType[]
+  formik: FormikProps<ImagePathTypes>
+  connectorRef: string
+  fieldName: string
+}
+
+export default function ServerlessArtifactoryRepository(
+  props: ServerlessArtifactoryRepositoryProps
+): React.ReactElement {
+  const { isReadonly, expressions, allowableTypes, formik, connectorRef, fieldName } = props
+
+  const { getString } = useStrings()
+  const [connectorRepos, setConnectorRepos] = useState<SelectOption[]>([])
+  const { accountId, projectIdentifier, orgIdentifier } = useParams<ProjectPathProps>()
+
+  const {
+    data: artifactRepoData,
+    loading: artifactRepoLoading,
+    refetch: getArtifactRepos,
+    error: artifactRepoError
+  } = useGetRepositoriesDetailsForArtifactory({
+    queryParams: {
+      connectorRef: connectorRef,
+      accountIdentifier: accountId,
+      orgIdentifier,
+      projectIdentifier,
+      repositoryType: 'generic'
+    },
+    lazy: true
+  })
+
+  useEffect(() => {
+    if (artifactRepoLoading) {
+      setConnectorRepos([{ label: 'Loading Repos...', value: 'Loading Repos...' }])
+    }
+    if ((artifactRepoError?.data as Failure)?.status === 'ERROR') {
+      const errorMessage = (artifactRepoError?.data as Failure)?.message as string
+      setConnectorRepos([{ label: errorMessage, value: errorMessage }])
+    } else if ((artifactRepoError?.data as Failure)?.status === 'FAILURE') {
+      const erroObj = (artifactRepoError?.data as Failure)?.errors?.[0]
+      const errorMessage =
+        erroObj?.fieldId && erroObj?.error ? `${erroObj?.fieldId} ${erroObj?.error}` : getString('somethingWentWrong')
+      setConnectorRepos([{ label: errorMessage, value: errorMessage }])
+    }
+  }, [artifactRepoLoading, artifactRepoError])
+
+  useEffect(() => {
+    if (getMultiTypeFromValue(connectorRef) === MultiTypeInputType.FIXED && !artifactRepoData) {
+      getArtifactRepos()
+    }
+    if (artifactRepoData) {
+      setConnectorRepos(map(artifactRepoData.data?.repositories, repo => ({ label: repo, value: repo })))
+    }
+  }, [artifactRepoData, connectorRef])
```

Comment:
Codacy has a fix for the issue: [React Hook useEffect has a missing dependency: 'getArtifactRepos'. Either include it or remove the dependency array.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232297)
```suggestion
  }, [artifactRepoData, connectorRef, getArtifactRepos])
```